pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -315,6 +317,11 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 		return obj, err
 	}
 
+	// infra was never adopted by a machine, just delete
```

Comment:
nit: can we add more context to this comment? Specifically around our observed machineset -> machine owner reference change, I think this would help external folks understand what is going on here more clearly